### DISCUSSION TITLE
How to set Clang platform?

### DISCUSSION THREAD

**Discussion by yh-sb** [xmake-io/xmake#4758] [Q&A] [answered]
There is `clang` option in `--toolchain`, but how to set clang platform? There is no `clang` in platforms option.
If I just set `xmake f --toolchain=clang` the project with boost package fails to build.

**Comment by waruqi** [comment, marked as answer]
no clang platform, you just set `--toolchain=clang`, but if you are using windows platform and installing package, you can only use msvc, do not switch toolchain, it does not support.

**Comment by yh-sb** [comment]
So, on Windows xrepo supports only MSVC?
Does this mean that I can't use Clang at all on Windows if my project uses some packages?

**Reply by waruqi** [reply]
Yes, switching toolchains on windows is too costly to maintain the package, and I don't have the time or more server resources to support it. It requires a lot of things to be considered, binary precompilation support, different compilers, runtime/mode/shared/static permutations for ci testing.

It will also cause to more complex configuration scripts for packages that are not easy to maintain

**Comment by Extra-Creativity** [comment]
I install clang by Visual Studio, which can use xmake packages normally (not clang-cl, but clang++).

**Reply by Extra-Creativity** [reply]
It seems that you cannot use clang on windows without gcc/msvc as stated in [stackoverflow](https://stackoverflow.com/a/63914740/15582103)

**Comment by waruqi** [comment]
> I install clang by Visual Studio, which can use xmake packages normally (not clang-cl, but clang++).

There are only a few packages maintained by xmake.lua, and some headeronly packages that support clang, but I don't maintain them on purpose. Also, there is no ci to test these packages. They are unreliable, and if they can be installed successfully, you can use them, but I don't maintain them for clang.

**Reply by Extra-Creativity** [reply]
I don't really get it; clang even doesn't use its own libraries on windows to be compatible with msvc/gcc, why is it unreliable?

**Reply by Extra-Creativity** [reply]
I mean, you can still use vs runtime, just change the toolchain

**Reply by waruqi** [reply]
We're only talking about toolchain switching for packages.

All package build scripts on windows are maintained only for msvc, there has never been any support or testing for clang.

**Reply by waruqi** [reply]
If you switch to clang, some package build scripts may or may not compile and pass, without any ci testing to guarantee it.

**Reply by Extra-Creativity** [reply]
Get it; I fail to install boost too :).

**Reply by waruqi** [reply]
Because boost's package configuration script never supported clang on windows, only for msvc.

**Comment by Extra-Creativity** [comment]
It seems that conan cannot cooperate happily with clang on windows for boost either, see https://github.com/conan-io/conan-center-index/issues/11970.
